src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -79,6 +80,11 @@ void CALLBACK cfApiFetchDataCallback(const CF_CALLBACK_INFO *callbackInfo, const
                               data.data(),
                               offset,
                               data.length());
+
+        LARGE_INTEGER progressCompleted;
+        progressCompleted.QuadPart = offset;
+
+        CfReportProviderProgress(callbackInfo->ConnectionKey, callbackInfo->TransferKey, callbackInfo->FileSize, progressCompleted);
```

Comment:
I'd say, please make a cfReportProviderProcess function akin to cfApiSendTransferInfo (or maybe move this into cfApiSendTransferInfo itself? but that'd force changing its interface a bit). Also I'd log CfReportProviderProgress failures just in case.

BTW, wouldn't you think they do that for you? I mean they know the total size (from the placeholder), the offset and the amount you just sent (provided in the CfExecute call)... but no, let's force people to mess with another piece of API just because. :-p

Reply:
@er-vin Progress is not refreshed if not calling this function.